aspnetcore/security/blazor/content-security-policy.md

```diff
@@ -0,0 +1,153 @@
+---
+title: Enforce a Content Security Policy for ASP.NET Core Blazor
+author: guardrex
+description: Learn how to use a Content Security Policy (CSP) in ASP.NET Core Blazor apps to help protect against Cross-Site Scripting (XSS) attacks.
+monikerRange: '>= aspnetcore-3.1'
+ms.author: riande
+ms.custom: mvc
+ms.date: 02/28/2020
+no-loc: [Blazor, SignalR]
+uid: security/blazor/content-security-policy
+---
+# Enforce a Content Security Policy for ASP.NET Core Blazor
+
+By [Javier Calvarro Nelson](https://github.com/javiercn) and [Luke Latham](https://github.com/guardrex)
+
+[!INCLUDE[](~/includes/blazorwasm-preview-notice.md)]
+
+[Cross-Site Scripting (XSS)](xref:security/cross-site-scripting) is a security vulnerability where an attacker places malicious client-side scripts into rendered content. A Content Security Policy (CSP) helps protect against XSS attacks by informing the browser of valid:
+
+* Sources for loaded content, including scripts, stylesheets, and images.
+* Actions taken by a page, including URL targets for form submissions.
+
+To use CSP, the developer specifies several CSP *directives* to browsers in one or more `Content-Security-Policy` headers or `<meta>` tags.
+
+Policies are evaluated while a document is loading. The browser inspects candidate sources and determines if the sources meet the requirements of the document's policies. When policy directives aren't met for a source, the browser blocks loading the resource. For example, consider a document with a policy that doesn't allow third-party scripts. When a document contains a script with a third-party origin in the `src` attribute of a `<script>` tag, the browser prevents the script from loading.
+
+CSP is supported in most modern desktop and mobile browsers, including Chrome, Edge, Firefox, Opera, and Safari. CSP is recommended for Blazor apps to help protect apps from XSS and leaking circuit IDs in Blazor Server apps. [Subresource Integrity (SRI)](https://developer.mozilla.org/docs/Web/Security/Subresource_Integrity) works in conjunction with CSPs.
+
+## Policy directives
+
+Minimally, specify the following directives and sources for Blazor apps. Add additional directives and sources as needed. The following directives are used in the [Apply the policy](#apply-the-policy) section of this article, where example security policies for Blazor WebAssembly and Blazor Server are provided:
+
+* [base-uri](https://developer.mozilla.org/docs/Web/HTTP/Headers/Content-Security-Policy/base-uri) &ndash; Restricts the URLs for a document's `<base>` tag.
+  * Specify `self` to indicate that the app's origin, including the scheme and port number, is a valid source.
+  * Supported by all browsers except IE.
+* [block-all-mixed-content](https://developer.mozilla.org/docs/Web/HTTP/Headers/Content-Security-Policy/block-all-mixed-content) &ndash; Prevents loading mixed HTTP and HTTPS content. Supported by all browsers except Edge/IE.
```

Comment:
It looks like this [is supported](https://caniuse.com/#search=block-all-mixed-content) in the re-imagination of Edge. Is it worth being more specific about that?